metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1295,27 +1296,53 @@ class MetalsLanguageServer(
         }.asJavaObject
       case ServerCommands.StartDebugAdapter() =>
         val args = params.getArguments.asScala
-        args match {
-          case Seq(param: JsonElement) =>
-            val session = for {
-              params <- Future.fromTry(param.as[b.DebugSessionParams])
-              server <- DebugServer.start(
-                params,
-                definitionProvider,
-                buildTargets,
-                buildServer
-              )
-            } yield {
-              cancelables.add(server)
-              DebugSession(server.sessionName, server.uri.toString)
-            }
-
-            session.asJavaObject
+        val debugSessionParamsParser = new JsonParser.Of[b.DebugSessionParams]
```

Comment:
Let's maybe move those parsers to an object, no need to recreate them each time.

Reply:
created `DebugParametersJsonParsers` with these parsers